### ISSUE TITLE
additional servers needed (missing python)

### ISSUE THREAD

**Issue by qmiwang** [krassowski/jupyterlab-lsp#470]
## Description

I have installed python 3.7.2 through pynv in a docker container.
And then, I have run the following commands to install jupyterlab and jupyterlab-lsp.
1. pip3 install --no-cache-dir jupyterlab     # jupyter lab 3.0.4 is installed
2. pip3 install jupyterlab-lsp==3.0.0
3.  pip3 install python-language-server[all]

Then run `jupyter labextension list `, I get

![image](https://user-images.githubusercontent.com/3363348/104594603-5b3cb180-56ac-11eb-80c2-b03657b2b618.png)

And config the language serve like this:

![image](https://user-images.githubusercontent.com/3363348/104595239-43196200-56ad-11eb-8131-3c6d94f83e1c.png)

However, it dose not work in the jupyter notebook. It shows "additional servers needed (missing python)"

![image](https://user-images.githubusercontent.com/3363348/104594738-8cb57d00-56ac-11eb-9617-45b84a081e6e.png)

Is there something wrong?

Thanks!

## Context

- Operating System and version: ubuntu 16.04, docker container
- Browser and version: chrome
- JupyterLab version: 3.0.4

</details>

**Comment by krassowski** [comment]
Thank you for opening this issue. I read it carefully but unfortunately, I cannot help much without having the [bug report template](https://github.com/krassowski/jupyterlab-lsp/issues/new?template=bug_report.md) filled in. It appears that you might have chosen a different template, or skipped fragments of it. In particular, missing parts of critical importance to your question are:

- Required: installed server extensions
- Troubleshoot Output
- Command Line Output

The template asks for information that we need in order to be able to help you. We know that this is a big ask, but this helps us diagnose issues quicker and gives us more time to improve this extension. If you have a problem filling it, please let us know and we will try to help (or improve the template if needed).

**Comment by krassowski** [comment]
@qmiwang do you use JupyterHub? What does `jupyter server extension list` produce for you?

**Comment by qmiwang** [comment]
> @qmiwang do you use JupyterHub? What does `jupyter server extension list` produce for you?

No, I didn't use Jupyter hub.  The output of `jupyter server extension list` is:

Config dir: /root/.jupyter

Config dir: /root/.pyenv/versions/3.7.2/etc/jupyter
    jupyter_kite enabled
    - Validating jupyter_kite...
      jupyter_kite 2.0.2 OK
    jupyter_lsp enabled
    - Validating jupyter_lsp...
      jupyter_lsp 1.0.0 OK
    jupyterlab enabled
    - Validating jupyterlab...
      jupyterlab 3.0.0 OK
    nbclassic enabled
    - Validating nbclassic...
      nbclassic  OK

Config dir: /usr/local/etc/jupyter

**Comment by krassowski** [comment]
Thanks! I see that you have `jupyter_kite` enabled. This extension cannot work side-by side with `jupyter-kite` as both use the same protocol and would conflict with each other. You would need to run:

`jupyter server extension disable jupyter_kite`

(possibly with  --sys-prefix or sudo, all depending on how you enabled it).

**Comment by qmiwang** [comment]
Thanks! I uninstall the jupyter-kit. But it still dosen't work.

I found there is a warning in the log of jupyter console:

![image](https://user-images.githubusercontent.com/3363348/104794927-e8375600-57e5-11eb-9c56-44f5abea555f.png)

**Comment by krassowski** [comment]
Thank you. It indicates that you have an old configuration of pyls  installed _somewhere_.  The current schema version is 2, but you have a file with version 1. Most likely you either:
- installed this extension a long ago and the configuration file remained, or
- you added this file manually
In either case you would want to find it and remove it. Use `jupyter --paths` to check where jupyter looks for config on your system (it will be likely 2-4 locations) and then check whether any of the files in the listed directories contains the pyls configuration in version one and then remove it and restart jupyterlab.

**Comment by qmiwang** [comment]
I changed the version 1 to 2 and restart the jupyterlab. It works now. Thanks!

**Comment by krassowski** [comment]
Happy it worked!

**Comment by qmiwang** [comment]
Unfortunately, the autocomplete is too slow. 

It takes 8~10s to show the completions, for example, "np."

Is there some solution?

**Comment by krassowski** [comment]
There is a pinned issue about performance. The general conclusion is that you may want to switch from pyls to Jedi language server. Also having an SSD helps. There are other fixes in the works - see the open PRs.

Reopening this as I'm thinking that maybe we should not raise on old schema version. What do you think @bollwyvl?

**Comment by dpriedel** [comment]
Is there a sample configuration to get the Jedi ls working?

**Comment by krassowski** [comment]
It was just added, and will be released with the next version today.

**Comment by dpriedel** [comment]
Great...Thanks !!

**Comment by krassowski** [comment]
@dpriedel the release with auto-detection of jedi server is now available; as noted in the release notes the jedi-language-server support is experimental; I recommend `jedi-language-server` in version 0.21 to prevent jedi-version incompatibility:

```
pip install jupyterlab-lsp==3.1.0 jupyter-lsp==1.1.0 jedi-language-server==0.21
```

**Comment by dpriedel** [comment]
Hi


Gave it a quick try and it seems to work Thanks!!!

Will spend more time later.


NOTE: I have my python in a non-standard location so I need to add it to 
my (Linux) PATH first.
